### PR TITLE
topologymanager: document topology manager policy options

### PR DESCRIPTION
**Pull request by PiotrProkop**
Enhancement link: https://github.com/kubernetes/enhancements/issues/3545
Implementation link: https://github.com/kubernetes/kubernetes/pull/112914

### CODE REVIEW

**Comment by PiotrProkop** [comment]
@sftim updated.

**Comment by klueska** [comment]
You need to retarget this to the dev-1.26 branch.

**Comment by fromanirh** [comment]
LGTM about the technical content from sig-node perspective.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -727,6 +730,15 @@ Each feature gate is designed for enabling/disabling a specific feature:
 - `TopologyManager`: Enable a mechanism to coordinate fine-grained hardware resource
   assignments for different components in Kubernetes. See
   [Control Topology Management Policies on a node](/docs/tasks/administer-cluster/topology-manager/).
+- `TopologyManagerPolicyAlphaOptions`: This allow fine-tuning of topology manager policies,
```
Comment (swatisehgal) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:733]:
```suggestion
- `TopologyManagerPolicyAlphaOptions`: Allow fine-tuning of topology manager policies,
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -727,6 +730,15 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
+  experimental, Alpha-quality options.
+  This feature gate guards *a group* of topology manager options whose quality level is alpha.
+  This feature gate will never graduate to beta or stable.
+- `TopologyManagerPolicyBetaOptions`: This allow fine-tuning of topology manager policies,
```
Comment (swatisehgal) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:737]:
```suggestion
- `TopologyManagerPolicyBetaOptions`: Allow fine-tuning of topology manager policies,
```

File: content/en/docs/tasks/administer-cluster/topology-manager.md [hunk 3]
```diff
@@ -213,6 +213,18 @@ reschedule the pod. It is recommended to use a Deployment with replicas to trigg
 the Pod.An external control loop could be also implemented to trigger a redeployment of pods
 that have the `Topology Affinity` error.
 
+### Topology manager policy options
+
+If the `prefer-closest-numa-nodes` policy option is specified, the `best-effort` and `restricted`
+policies will favor sets of NUMA nodes with shorter distance between nodes when making admission decisions.
```
Comment (swatisehgal) [inline content/en/docs/tasks/administer-cluster/topology-manager.md:219]:
```suggestion
policies will favor sets of NUMA nodes with shorter distance between them when making admission decisions.
```

File: content/en/docs/tasks/administer-cluster/topology-manager.md [hunk 4]
```diff
@@ -213,6 +213,18 @@ reschedule the pod. It is recommended to use a Deployment with replicas to trigg
...
+You can enable this option by adding `prefer-closest-numa-nodes=true` to the Topology Manager policy options.
+By default, without this option, Topology Manager align resources on either a single NUMA node or
```
Comment (swatisehgal) [inline content/en/docs/tasks/administer-cluster/topology-manager.md:221]:
```suggestion
By default, without this option, Topology Manager aligns resources on either a single NUMA node or
```

**Review by swatisehgal — COMMENTED** [review]
Noticed a few minor NITs but looks good overall from node-perspective.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 1 above)
Comment (PiotrProkop) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: fixed

File: content/en/docs/tasks/administer-cluster/topology-manager.md (hunk 3 above)
Comment (PiotrProkop) [reply content/en/docs/tasks/administer-cluster/topology-manager.md]: fixed

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 2 above)
Comment (PiotrProkop) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: fixed

File: content/en/docs/tasks/administer-cluster/topology-manager.md (hunk 4 above)
Comment (PiotrProkop) [reply content/en/docs/tasks/administer-cluster/topology-manager.md]: fixed

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 5]
```diff
@@ -727,6 +730,15 @@ Each feature gate is designed for enabling/disabling a specific feature:
 - `TopologyManager`: Enable a mechanism to coordinate fine-grained hardware resource
   assignments for different components in Kubernetes. See
   [Control Topology Management Policies on a node](/docs/tasks/administer-cluster/topology-manager/).
+- `TopologyManagerPolicyAlphaOptions`: Allow fine-tuning of topology manager policies,
+  experimental, Alpha-quality options.
+  This feature gate guards *a group* of topology manager options whose quality level is alpha.
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:735]: So the features behind this gate are designed to be a mystery, right?

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 6]
```diff
@@ -727,6 +730,15 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
+  This feature gate will never graduate to beta or stable.
+- `TopologyManagerPolicyBetaOptions`: Allow fine-tuning of topology manager policies,
+  experimental, Beta-quality options.
+  This feature gate guards *a group* of topology manager options whose quality level is alpha.
+  This feature gate will never graduate to stable.
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:740]: Same here. We don't want users to know what are behind the gate, and we want them to try them out?

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 5 above)
Comment (PiotrProkop) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:735]: section added, please let me know if it helps 😄

**Review by fromanirh — COMMENTED** [review]
/lgtm
technical content from sig-node perspective

**Comment by krol3** [comment]
Hi @PiotrProkop ! This PR needs a doc review by **Mon Nov 28th** to get this into the release. Please reach out to required SIGs to get their review. Thank you!

**Comment by PiotrProkop** [comment]
@krol3 I've got `lgtm` label from `sig/node`

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 6 above)
Comment (klueska) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:740]: The specific options gated by this feature gate is described in the relevant documentation for the TopologyManager itself, e.g. 
`content/en/docs/tasks/administer-cluster/topology-manager.md`.

There is no precedent for listing out the specific options here (if that is what you are suggesting).

**Comment by klueska** [comment]
/lgtm
(as [sig-node-approver](https://github.com/kubernetes/kubernetes/blob/master/OWNERS_ALIASES#L225...L232))

**Review by sftim — COMMENTED** [review]
We should add a list of alpha options and beta options into a new page within https://kubernetes.io/docs/reference/node/
(and not https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/ which is a task page; task pages are the wrong place to put reference docs)

Although that change is really important, for me it doesn't block merging this PR.

@PiotrProkop / @klueska would you be willing to file a docs issue about adding that page, so we don't forget? Once the work is at least **tracked**, I think this is good to go.

**Comment by PiotrProkop** [comment]
@sftim https://github.com/kubernetes/website/issues/38119. done 😄

**Pull request merged**